### PR TITLE
update project deps and switch to non package config for the main pro…

### PR DESCRIPTION
**Pull request by aromanov-adi**
- update dependencies versions
- set package-mode = false for the main project config, so poetry is happy and not trying to install the project as a package

### CODE REVIEW

File: {{cookiecutter.project_folder}}/pyproject.toml [hunk 1]
```diff
@@ -4,31 +4,31 @@ version = "{{ cookiecutter.version }}"
 description = "{{ cookiecutter.short_description }}"
 authors = ["{{ cookiecutter.full_name }} <{{ cookiecutter.email }}>"]
 readme = "README.md"
-packages = [{include =  "{{ cookiecutter.module_name }}", from = "{{ cookiecutter.namespace }}"}]
+package-mode = false
```
Comment (aromanov-adi) [inline {{cookiecutter.project_folder}}/pyproject.toml:7]: this is arguable thing, if the idea is to have a project as a dependency of other projects then we may need to keep the original value.
from my perspective current project is an independent project

Comment (parthbarhan-adi) [reply {{cookiecutter.project_folder}}/pyproject.toml:7]: I agree. I don't see this project as a dependency for other projects

File: {{cookiecutter.project_folder}}/pyproject.toml [hunk 2]
```diff
@@ -4,31 +4,33 @@ version = "{{ cookiecutter.version }}"
...
 python = ">={{ cookiecutter.python_version }}, <3.13"
 {%- if cookiecutter.use_jupyterlab == "y" %}
 jupyterlab = "^3.0.0"
 {%- endif %}
-pydantic = "^1.10.7"
-fastapi = "^0.112.2"
+pydantic = "^2.9.2"
+fastapi = "^0.115.2"
 typer = "^0.12.3"
+uvicorn = "^0.32.0"
```
Comment (parthbarhan-adi) [inline {{cookiecutter.project_folder}}/pyproject.toml:17]: Do we need uvicorn? I think based on the type of project being built the user can add uvicorn later.

**Review by parthbarhan-adi — APPROVED** [review]
Looks good!

**Pull request merged**